Kernel/FileSystem/FileSystem.cpp

```diff
@@ -81,4 +81,12 @@ void FS::set_block_size(size_t block_size)
     m_block_size = block_size;
 }
 
+void FS::set_fragment_size(size_t fragment_size)
+{
+    VERIFY(fragment_size > 0);
```

Comment:
Well, if I understood correctly, fragments are chunks used for smaller files or remainder of files to be allocated to to not waste full blocks. So, if the fragment size would be zero, how would those fragments work then?
Or would a 0 here mean "unsupported"/"use full blocks"?

Reply:
My understanding is that other implementations of ext2 and ext3 don't actually use this feature in practice. So I'm not sure it's something we need to care about either.

Either way, I'm just asking if it's appropriate to panic the kernel if someone calls this API with `0` :)

Reply:
If it's unused except for being queried by statvfs, then we don't need to crash the kernel. Though as this is part of the generic `FileSystem.cpp`, and may get used in a future implementation of ext4, I think it's appropriate to crash the kernel at disk initialization rather than later when the fragment size is used for file allocation, as this API should only be called at disk initialization.
Though I might just be future-proofing a bit too strongly here.

Reply:
Or do you rather mean that we shouldn't crash if a specific ext2 formatting tool puts 0 as a fragment size onto the filesystems header?

Reply:
Ehh, we're probably overthinking this. It's fine the way it is, and we'll notice immediately if it's a problem :)